src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -314,6 +403,194 @@ private void ResolveAsymmetricAlgorithm(AsymmetricSecurityKey key, string algori
         }
 #endif
 
+        private RSAParameters CreateRsaParametersFromJsonWebKey(JsonWebKey webKey, bool willCreateSignatures)
+        {
+            if (webKey == null)
+                throw LogHelper.LogArgumentNullException(nameof(webKey));
+
+            if (webKey.N == null || webKey.E == null)
+                throw LogHelper.LogException<ArgumentException>(LogMessages.IDX10700, webKey);
+
+            RSAParameters parameters;
+            if (willCreateSignatures)
+            {
+                if (webKey.D == null || webKey.DP == null || webKey.DQ == null || webKey.QI == null || webKey.P == null || webKey.Q == null)
+                    throw LogHelper.LogException<ArgumentNullException>(LogMessages.IDX10702, webKey);
+
+                parameters = new RSAParameters()
+                {
+                    D = Base64UrlEncoder.DecodeBytes(webKey.D),
+                    DP = Base64UrlEncoder.DecodeBytes(webKey.DP),
+                    DQ = Base64UrlEncoder.DecodeBytes(webKey.DQ),
+                    Exponent = Base64UrlEncoder.DecodeBytes(webKey.E),
+                    Modulus = Base64UrlEncoder.DecodeBytes(webKey.N),
+                    InverseQ = Base64UrlEncoder.DecodeBytes(webKey.QI),
+                    P = Base64UrlEncoder.DecodeBytes(webKey.P),
+                    Q = Base64UrlEncoder.DecodeBytes(webKey.Q)
+                };
+            }
+            else
+            {
+                parameters = new RSAParameters()
+                {
+                    Exponent = Base64UrlEncoder.DecodeBytes(webKey.E),
+                    Modulus = Base64UrlEncoder.DecodeBytes(webKey.N),
+                };
+            }
+            return parameters;
+        }
+
+        private void CreateECDsaFromJsonWebKey(JsonWebKey webKey, bool willCreateSignatures)
+        {
+            if (webKey == null)
+                throw LogHelper.LogArgumentNullException(nameof(webKey));
+
+            if (webKey.Crv == null)
+                throw LogHelper.LogArgumentNullException(nameof(webKey.Crv));
+
+            if (webKey.X == null)
+                throw LogHelper.LogArgumentNullException(nameof(webKey.X));
+
+            if (webKey.Y == null)
+                throw LogHelper.LogArgumentNullException(nameof(webKey.Y));
+
+            GCHandle keyBlobHandle = new GCHandle();
```

Comment:
Do we need to 'new GCHandle()' here? Why not just 'null'; keyBlobHandle is set with GCHandle.Alloc(...) below.